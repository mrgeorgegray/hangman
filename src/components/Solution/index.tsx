/** @jsxImportSource @emotion/react */
import React from "react";
import { css, useTheme } from "@emotion/react";

import { GameStatus } from "../../context/Game/state";
interface SolutionProps {
  status: GameStatus;
  text: string;
}

const Solution: React.FC<SolutionProps> = ({ status, text }) => {
  const { breakpoints, colors, fontSize, space } = useTheme();
  let textColor: string;

  switch (status) {
    case GameStatus.Won:
      textColor = colors.success;
      break;
    case GameStatus.Lost:
      textColor = colors.error;
      break;
    default:
      textColor = colors.text;
      break;
  }

  return (
    <div
      css={css`
        color: ${textColor};
        font-size: ${fontSize[3]}px;
        letter-spacing: 4px;
        line-height: 1.2;
        margin-bottom: ${space[3]};

        @media (min-width: ${breakpoints.md}) {
          font-size: ${fontSize[4]}px;
        }
      `}
    >
      {text}
    </div>
  );
};

export default Solution;
