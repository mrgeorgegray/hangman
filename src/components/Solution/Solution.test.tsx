import React from "react";
import { render } from "@testing-library/react";

import Solution from ".";

describe("<Solution />", () => {
  it("renders", () => {
    const { container } = render(<Solution text="s_lution" />);
    expect(container).toMatchSnapshot();
  });
});
