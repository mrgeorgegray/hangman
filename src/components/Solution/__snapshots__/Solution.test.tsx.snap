// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Solution /> renders 1`] = `
<div>
  <div
    class="css-1am8y5a-Solution"
  >
    s_lution
  </div>
</div>
`;
